app/services/hyrax/tika_file_characterization_service.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+# Class for Apache Tika based file characterization service
+# defines the Apache Tika based characterization service a ValkyrieFileCharacterization service
+# @since 3.0.0
+module Hyrax
+  class TikaFileCharacterizationService
+    attr_reader :file_node, :persister
+    def initialize(file_node:, persister:)
+      @file_set = file_node
+      @persister = persister
+    end
+
+    # characterizes the file_node (FileSet) passed into this service
+    # @return [FileSet]
+    # @example characterize a file and persist the changes
+    #   Valkyrie::FileCharacterizationService.for(file_node, persister).characterize
+    def characterize
+      result = JSON.parse(json_output)
+      @file_characterization_attributes = {
+        width: result['tiff:ImageWidth'],
```

Comment:
is this only going to work for TIFF sources?  What about JP2 or Jpeg?

Reply:
Yes, works for other image types as well.